### ISSUE TITLE
Empty ORC file is not supported

### ISSUE THREAD

**Issue by wyukawa** [prestodb/presto#9354]
Hi,

I use presto 0.189 + HDP 2.5.3 + hive connector.

```
select * from ... where yyyymmdd='20171115' limit 100
```

if ```hdfs://nameservice1/apps/hive/warehouse/.../yyyymmdd=20171115/000007_0``` is empty file, the following error occurs.

This error doesn't occur in presto 0.188.
We have many tables which contains empty file.
If presto skips empty file, it would be nice.

```
com.facebook.presto.spi.PrestoException: Error opening Hive split hdfs://nameservice1/apps/hive/warehouse/.../yyyymmdd=20171115/000007_0 (offset=0, length=0): size must be at least 1
	at com.facebook.presto.hive.orc.OrcPageSourceFactory.createOrcPageSource(OrcPageSourceFactory.java:176)
	at com.facebook.presto.hive.orc.OrcPageSourceFactory.createPageSource(OrcPageSourceFactory.java:112)
	at com.facebook.presto.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:133)
	at com.facebook.presto.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:88)
	at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:44)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:56)
	at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:224)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:337)
	at com.facebook.presto.operator.Driver.lambda$processFor$6(Driver.java:241)
	at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:614)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:235)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:622)
	at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at com.facebook.presto.execution.executor.LegacyPrioritizedSplitRunner.process(LegacyPrioritizedSplitRunner.java:23)
	at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:492)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: size must be at least 1
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
	at com.facebook.presto.orc.AbstractOrcDataSource.<init>(AbstractOrcDataSource.java:57)
	at com.facebook.presto.hive.orc.HdfsOrcDataSource.<init>(HdfsOrcDataSource.java:46)
	at com.facebook.presto.hive.orc.OrcPageSourceFactory.createOrcPageSource(OrcPageSourceFactory.java:162)
	... 17 more
```

Thanks

**Comment by findepi** [comment]
This was a conscious decision introduced in #9316. Is empty file a valid ORC file?

cc @haozhun

**Comment by wyukawa** [comment]
Thank you for comment.

How can I check if empty file is a valid ORC file?

I execute orcfiledump, the following error occurs.
```
$ hive --orcfiledump /apps/hive/warehouse/.../yyyymmdd=20171115/000007_0
Processing data file /apps/hive/warehouse/.../yyyymmdd=20171115/000007_0 [length: 0]
Exception in thread "main" java.lang.IndexOutOfBoundsException
        at java.nio.Buffer.checkIndex(Buffer.java:540)
        at java.nio.HeapByteBuffer.get(HeapByteBuffer.java:139)
        at org.apache.hadoop.hive.ql.io.orc.ReaderImpl.extractFileTail(ReaderImpl.java:383)
        at org.apache.hadoop.hive.ql.io.orc.ReaderImpl.<init>(ReaderImpl.java:321)
        at org.apache.hadoop.hive.ql.io.orc.OrcFile.createReader(OrcFile.java:241)
        at org.apache.hadoop.hive.ql.io.orc.FileDump.getReader(FileDump.java:228)
        at org.apache.hadoop.hive.ql.io.orc.FileDump.printMetaDataImpl(FileDump.java:282)
        at org.apache.hadoop.hive.ql.io.orc.FileDump.printMetaData(FileDump.java:261)
        at org.apache.hadoop.hive.ql.io.orc.FileDump.main(FileDump.java:127)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.hadoop.util.RunJar.run(RunJar.java:233)
        at org.apache.hadoop.util.RunJar.main(RunJar.java:148)
```

I don't know why empty file is created...

**Comment by haozhun** [comment]
@findepi  Apparently empty ORC file is valid: https://issues.apache.org/jira/browse/ORC-162

@dain Opinions?

**Comment by wyukawa** [comment]
Actually, we often have empty ORF files in hive tables.
So if presto can handle  empty ORC files, we would be very happy.

**Comment by wyukawa** [comment]
@findepi @haozhun @dain Do you have a plan to fix this issue?

**Comment by electrum** [comment]
Empty ORC files previously worked by accident. However, since Hive considers these valid, we should, too. I'll put up a fix.

**Comment by electrum** [comment]
This will be fixed in the next release, 0.192.

**Comment by wyukawa** [comment]
Thanks!!!!!!!!!!!
